src/main/java/backend/resource/TurboLabel.java

```diff
@@ -211,4 +224,138 @@ public int hashCode() {
     public int compareTo(TurboLabel o) {
         return actualName.compareTo(o.getActualName());
     }
+
+    /**
+     * Returns the TurboLabel in labels that matches labelName
+     * Assumption: the labelName matches exactly 1 TurboLabel
+     *
+     * @param labels
+     * @param labelName
+     * @return
+     */
+    public static TurboLabel getMatchingTurboLabel(List<TurboLabel> labels, String labelName) {
+        assert labels.stream()
+                .filter(label -> label.getActualName().equals(labelName))
+                .findFirst()
+                .isPresent();
+        return labels.stream()
+                .filter(label -> label.getActualName().equals(labelName))
+                .findFirst()
+                .get();
+    }
+
+    public static List<String> getLabelsNameList(List<TurboLabel> labels) {
+        return labels.stream()
+                .map(TurboLabel::getActualName)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of label names that matches a query
+     * Uses partial matching and matching is case-insensitive
+     * @param repoLabels
+     * @param queryLabelName
+     * @return
+     */
+    public static List<String> filterByPartialName(List<String> repoLabels, String queryLabelName) {
+        return repoLabels
+                .stream()
+                .filter(name -> Utility.containsIgnoreCase(
+                    new TurboLabel("", name).getSimpleName(), queryLabelName))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of label names that belongs to a group and matches a query
+     * Uses partial matching and matching is case-insensitive
+     * @param repoLabels
+     * @param queryLabelGroup
+     * @return
+     */
+    public static List<String> filterByPartialGroupName(List<String> repoLabels, String queryLabelGroup) {
+        if (queryLabelGroup.isEmpty()) return repoLabels;
+
+        return repoLabels
+                .stream()
+                .filter(name -> {
+                    TurboLabel newLabel = new TurboLabel("", name);
+                    if (newLabel.isInGroup()) {
+                        return Utility.containsIgnoreCase(newLabel.getGroupName(), queryLabelGroup);
+                    }
+                    return false;
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the first label that matches the keyword
```

Comment:
If you say 'first match', then it is based on the sort order of the input? Still fine. Just an observation.